### PR TITLE
feat[configuration-validation]: Adds configuration-validation package

### PR DESCRIPTION
**Pull request by manueltanzi-okta**
Adds a new package that exposes the functions to validate the configuration of the other SDKs.

This PR is broken into the following commits:
* [`a7376b5`](https://github.com/okta/okta-oidc-js/pull/316/commits/a7376b54b0b8b9066953fcbd2b63b721844d414d) - Adds configuration library
* [`3bf3a8d`](https://github.com/okta/okta-oidc-js/pull/316/commits/3bf3a8d36ae05c1f2b5bec6619ea99667d8bc85e) and [`d3af902`](https://github.com/okta/okta-oidc-js/pull/316/commits/d3af902a815628a79908e94415d35c714011ca7e) - Adds tests and eslint
* [`d965df0`](https://github.com/okta/okta-oidc-js/pull/316/commits/d965df0fb14438502d7dd3baa46b476f3b0d3e35) - Basic API documentation

### Reviewers
@robertdamphousse-okta, @jmelberg-okta, @manueltanzi-okta

### CODE REVIEW

**Comment by manueltanzi-okta** [comment]
Should we move the unit tests here?

File: packages/configuration-validation/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@okta/configuration-validation",
+  "version": "1.0.0",
```
Comment (manueltanzi-okta) [inline packages/configuration-validation/package.json]: Should we start from 0.0.1 like in the jwt-verifier?

Comment (jmelberg-okta) [reply packages/configuration-validation/package.json]: We can probably start with an `0.1.0` version, if we want to offer this as a pre-release.

I have a strong feeling that developers are **not** going to consume this library directly, deferring to our integrations.

**Comment by manueltanzi-okta** [comment]
Additions LGTM! Should we link this package in the README?

File: packages/configuration-validation/LICENSE [hunk 2]
```diff
@@ -0,0 +1,187 @@
+The Okta software accompanied by this notice is provided pursuant to the
+following terms:
+
+Copyright © 2015-2017, Okta, Inc.
```
Comment (robertjd) [inline packages/configuration-validation/LICENSE]: Please update to be 2018-Present

**Pull request merged**